### PR TITLE
Blocks diff (chapter 1)

### PR DESCRIPTION
**Pull request by pelikhan**
This is a work in progress for blocks diff view. It currently relies on blockids. We will need a structural diff that is ID agnostic in the future.

test link: https://arcade.makecode.com/app/d2016dd681199f057223709244b9e7807232fa95-a955512428

image

### CODE REVIEW

File: theme/blockly-core.less [hunk 1]
```diff
@@ -211,6 +211,10 @@ text.blocklyCheckbox {
 }
 
 
+.blocklyDisabled .blocklyNonEditableText .blocklyBlockBackground {
```
Comment (pelikhan) [inline theme/blockly-core.less]: this change needs to be upstreamed in pxt-blockly

Comment (pelikhan) [reply theme/blockly-core.less]: https://github.com/microsoft/pxt/issues/5932

**Comment by pelikhan** [comment]
@ganicke the docs will need a layer of polish...

File: pxtblocks/blocklydiff.ts [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+        function normalizeAttributes(e: Element) {
+            e.removeAttribute("id");
+            e.removeAttribute("x");
+            e.removeAttribute("y");
+            e.removeAttribute("deletable");
+            e.removeAttribute("editable");
+            e.removeAttribute("movable")
+        }
+
+        function visDom(el: Element, f: (e: Element) => void) {
```
Comment (jwunderl) [inline pxtblocks/blocklydiff.ts]: is there a better name we could use for this? / could this just be done as `el.queryselectorAll("*").forEach(f)` or something of the sort?

File: pxtblocks/blocklydiff.ts [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+                const nextw = ws.getBlockById(dids[next.id]) || ws.getBlockById(next.id);
+                if (nextw) {
+                    log(`next ${b.id}->${wb.toDevString()}: ${nextw.toDevString()}`)
+                    wb.nextConnection.connect(nextw.previousConnection);
+                }
+            }
+        }
+
+        function markUsed(b: Blockly.Block) {
+            (<any>b).___used = true;
```
Comment (mmoskal) [inline pxtblocks/blocklydiff.ts]: I would go with `_pxt_used` or something

File: pxtlib/service.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ namespace ts.pxtc {
     export const TS_CONTINUE_TYPE = "continue_keyword";
     export const TS_OUTPUT_TYPE = "typescript_expression";
     export const PAUSE_UNTIL_TYPE = "pxt_pause_until";
+    export const COLLAPSED_BLOCK = "__collapsed_block"
```
Comment (mmoskal) [inline pxtlib/service.ts]: `pxt_collapsed_block`?

**Pull request merged**